File: test_db_connection.py
import os
from dotenv import load_dotenv
import mysql.connector
from mysql.connector import Error

# Carica le variabili d'ambiente dal file .env
load_dotenv()

# Configurazione della connessione
config = {
    'host': os.getenv('DB_HOST'),
    'user': os.getenv('DB_USER'),
    'password': os.getenv('DB_PASSWORD'),
    'port': os.getenv('DB_PORT'),
    'database': os.getenv('DB_NAME')
}

print('Tentativo di connessione al database...')

try:
    connection = mysql.connector.connect(**config)
    if connection.is_connected():
        print('✅ Connessione al database riuscita!')
        print(f"Server: {config['host']}:{config['port']}")
        print(f"Database: {config['database']}")
except Error as err:
    print('❌ Errore di connessione:', err)
    print('Dettagli connessione:')
    print(f"Host: {config['host']}")
    print(f"Porta: {config['port']}")
    print(f"Database: {config['database']}")
    print(f"Utente: {config['user']}")
finally:
    if 'connection' in locals() and connection.is_connected():
        connection.close()

File: test-db-connection.js
require('dotenv').config();
const mysql = require('mysql2');

const connection = mysql.createConnection({
  host: process.env.DB_HOST,
  user: process.env.DB_USER,
  password: process.env.DB_PASSWORD,
  port: process.env.DB_PORT,
  database: process.env.DB_NAME
});

console.log('Tentativo di connessione al database...');

connection.connect((err) => {
  if (err) {
    console.error('❌ Errore di connessione:', err);
    console.error('Dettagli connessione:');
    console.error(`Host: ${process.env.DB_HOST}`);
    console.error(`Porta: ${process.env.DB_PORT}`);
    console.error(`Database: ${process.env.DB_NAME}`);
    console.error(`Utente: ${process.env.DB_USER}`);
  } else {
    console.log('✅ Connessione al database riuscita!');
    console.log(`Server: ${connection.config.host}:${connection.config.port}`);
    console.log(`Database: ${connection.config.database}`);
  }
  
  connection.end();
});
